DuckDB.NET.Data/DuckDBSchema.cs

```diff
@@ -0,0 +1,202 @@
+                { DuckDbMetaDataCollectionNames.Columns, 4, 4 },
+                { DuckDbMetaDataCollectionNames.ForeignKeys, 4, 3 },
+                { DuckDbMetaDataCollectionNames.Indexes, 4, 3 },
+            }
+        };
+
+    private static DataTable GetRestrictions() =>
+        new(DbMetaDataCollectionNames.Restrictions)
+        {
+            Columns =
+            {
+                { "CollectionName", typeof(string) },
+                { "RestrictionName", typeof(string) },
+                { "RestrictionDefault", typeof(string) },
+                { "RestrictionNumber", typeof(int) }
+            },
+            Rows =
+            {
+                { DuckDbMetaDataCollectionNames.Tables, "Catalog", "table_catalog", 1 },
+                { DuckDbMetaDataCollectionNames.Tables, "Schema", "table_schema", 2 },
+                { DuckDbMetaDataCollectionNames.Tables, "Table", "table_name", 3 },
+                { DuckDbMetaDataCollectionNames.Tables, "TableType", "table_type", 4 },
+
+                { DuckDbMetaDataCollectionNames.Columns, "Catalog", "table_catalog", 1 },
+                { DuckDbMetaDataCollectionNames.Columns, "Schema", "table_schema", 2 },
+                { DuckDbMetaDataCollectionNames.Columns, "Table", "table_name", 3 },
+                { DuckDbMetaDataCollectionNames.Columns, "Column", "column_name", 4 },
+
+                { DuckDbMetaDataCollectionNames.ForeignKeys, "Catalog", "constraint_catalog", 1 },
+                { DuckDbMetaDataCollectionNames.ForeignKeys, "Schema", "constraint_schema", 2 },
+                { DuckDbMetaDataCollectionNames.ForeignKeys, "Table", "table_name", 3 },
+                { DuckDbMetaDataCollectionNames.ForeignKeys, "Constraint", "constraint_name", 4 },
+
+                { DuckDbMetaDataCollectionNames.Indexes, "Catalog", "constraint_catalog", 1 },
+                { DuckDbMetaDataCollectionNames.Indexes, "Schema", "constraint_schema", 2 },
+                { DuckDbMetaDataCollectionNames.Indexes, "Table", "table_name", 3 },
+                { DuckDbMetaDataCollectionNames.Indexes, "Constraint", "constraint_name", 4 },
+            },
+        };
+
+    private static DataTable GetReservedWords(DuckDBConnection connection)
+    {
+        using var command = connection.CreateCommand();
+        command.CommandText = "SELECT keyword_name as ReservedWord FROM duckdb_keywords() WHERE keyword_category = 'reserved'";
+        command.CommandType = CommandType.Text;
+        return GetDataTable(DbMetaDataCollectionNames.ReservedWords, command);
+    }
+
+    private static DataTable GetTables(DuckDBConnection connection, string?[]? restrictionValues)
+    {
+        const string query = "SELECT table_catalog, table_schema, table_name, table_type FROM information_schema.tables";
+
+        using var command = BuildCommand(connection, query, restrictionValues, true,
+            ["table_catalog", "table_schema", "table_name", "table_type"]);
```

Comment:
If the numbers above refer to length of these arrays, can you extract these arrays somewhere and refer to their length in `GetMetaDataCollections` ?

Reply:
> If the numbers above refer to the length of these arrays

Yes,

> can you extract these arrays somewhere and refer to their Length in GetMetaDataCollections

I could, try, but it would just complicate the code, I afraid, without providing any benefit.